server/internal/storage/gorm/client.go

```diff
@@ -37,124 +36,99 @@ type Client struct {
 var mutex sync.Mutex
 var disableMutex bool
 
-func mylock() {
+func lock() {
```

Comment:
I added this lock when I noticed problems due to concurrent access to the sqlite backend. I don't think our integration tests check this (yet), but when the demo scripts run bulk uploads (and other concurrent tasks), the server gets parallel writes - so this lock serializes them. As I understand it (and discussed elsewhere in existing comments), Postgres doesn't need this, so we only enable locks for SQLite backends.

... so I wouldn't want to remove this lock without tests showing that concurrent writes to SQLite are safe without it (and we should probably also validate the assumption that concurrent writes are fine with Postgres).